Add launcher example (#65)
jtpio/jupyterlab-extension-examples: launcher/README.md

commands.addCommand(CommandIDs.createNew, {
  label: args => (args['isPalette'] ? 'New Python File' : 'Python File'),
  caption: 'Create a new Python file',
  iconClass: args => (args['isPalette'] ? '' : ICON_CLASS),
  execute: async args => {
    // Get the directory in which the Python file must be created;
    // otherwise take the current filebrowser directory
    let cwd = args['cwd'] || browserFactory.defaultBrowser.model.path;

    // Create a new untitled python file
    const model = await commands.execute('docmanager:new-untitled', {
      path: cwd,
      type: 'file',
      ext: 'py'
    });

    // Open the newly created file with the 'Editor'
    return commands.execute('docmanager:open', {
      path: model.path,
      factory: FACTORY
    });
  }
});
```

To link that command to the JupyterLab launcher, the `ILauncher` interface needs to be passed to the `activate`
extension function. As that interface is provided by the `@jupyterlab/launcher` package, it needs first to be installed:

```bash
jlpm add @jupyterlab/launcher
```

Then you can use it in the extension by importing it:

```ts
// src/index.ts#L10-L10

import { ILauncher } from '@jupyterlab/launcher';
```

> And finally you can request it as extension dependency:

Reviewer: Or?

```suggestion
And finally you can add it to the list of dependencies:
```

```ts
// src/index.ts#L22-L33

const extension: JupyterFrontEndPlugin<void> = {
  id: '@jupyterlab-examples/launcher',
  autoStart: true,
  optional: [ILauncher, IMainMenu, ICommandPalette],

>   requires: [IFileBrowserFactory],

Reviewer: Should we position `requires` first, and then `optional`? So the order is the same as the parameters in the `activate` function.

Reviewer: Not sure there is a clear pattern for that though (also in core lab).

Author: I agree with you. There is no clear pattern to have the attribute in alphabetical order. And as the required arguments come before the optional ones, it makes sense to list them in that order.

  activate: (
    app: JupyterFrontEnd,
    browserFactory: IFileBrowserFactory,
    launcher: ILauncher | null,
    menu: IMainMenu | null,
    palette: ICommandPalette | null
  ) => {
```

> In this example, the `ILauncher` interface is requested as optional dependency and not as classical dependency. This allow other application without launcher to be able

Reviewer: We could directly use the same term with `require`:

```suggestion
In this example, the `ILauncher` interface is requested as optional dependency and not as a required dependency. This lets other applications without a launcher be able
```
